src/components/Axis/VerticalAxis.tsx

```diff
@@ -30,36 +37,70 @@ export default function VerticalAxis(props: AxisRendererProps) {
   const ticks = useBBoxObserver();
 
   function getTickPosition(value: number) {
+    const { x1, x2, textPosition } = GetTickX();
     return {
       line: {
         y1: value,
         y2: value,
-        x2: !isRight ? -5 : 5,
+        x1: x1,
+        x2: x2,
       },
-      text: { y1: value, x1: isRight ? 10 : -10 },
+      text: { y1: value, x1: isRight ? textPosition : -textPosition },
     };
   }
-
+  function GetTickX() {
+    const x = isRight ? primaryTickLength : -1 * primaryTickLength;
+    switch (tickPosition) {
+      case 'center':
+        return {
+          x1: x / 2,
+          x2: -x / 2,
+          textPosition: 3 + primaryTickLength / 2,
```

Comment:
Can you put the "3" in a constant (textOffset ?) and use it everywhere? In the future it will be an option